Code/Framework/AzCore/AzCore/DOM/DomPatch.h

```diff
@@ -0,0 +1,191 @@
+{
+    //! A patch operation that represents an atomic operation for mutating or validating a Value.
+    //! PatchOperations can be created with helper methods in Patch. /see Patch
+    class PatchOperation final
+    {
+    public:
+        using PatchOutcome = AZ::Outcome<void, AZStd::string>;
+
+        //! The operation to perform.
+        enum class Type
+        {
+            Add, //!< Inserts or replaces the value at DestinationPath with Value
+            Remove, //!< Removes the entry at DestinationPath
+            Replace, //!< Replaces the value at DestinationPath with Value
+            Copy, //!< Copies the contents of SourcePath to DestinationPath
+            Move, //!< Moves the contents of SourcePath to DestinationPath
+            Test //!< Ensures the contents of DestinationPath match Value or fails, performs no mutations
+        };
+
+        PatchOperation() = default;
+        PatchOperation(const PatchOperation&) = default;
+        PatchOperation(PatchOperation&&) = default;
+
+        explicit PatchOperation(Path destinationPath, Type type, Value value);
+        explicit PatchOperation(Path destionationPath, Type type, Path sourcePath);
+        explicit PatchOperation(Path path, Type type);
+
+        static PatchOperation AddOperation(Path destinationPath, Value value);
+        static PatchOperation RemoveOperation(Path pathToRemove);
+        static PatchOperation ReplaceOperation(Path destinationPath, Value value);
+        static PatchOperation CopyOperation(Path destinationPath, Path sourcePath);
+        static PatchOperation MoveOperation(Path destinationPath, Path sourcePath);
+        static PatchOperation TestOperation(Path testPath, Value value);
+
+        PatchOperation& operator=(const PatchOperation&) = default;
+        PatchOperation& operator=(PatchOperation&&) = default;
+
+        bool operator==(const PatchOperation& rhs) const;
+        bool operator!=(const PatchOperation& rhs) const;
+
+        Type GetType() const;
+        void SetType(Type type);
+
+        const Path& GetDestinationPath() const;
+        void SetDestinationPath(Path path);
+
+        const Value& GetValue() const;
+        void SetValue(Value value);
+
+        const Path& GetSourcePath() const;
+        void SetSourcePath(Path path);
+
+        AZ::Outcome<Value, AZStd::string> Apply(Value rootElement) const;
+        PatchOutcome ApplyInPlace(Value& rootElement) const;
+
+        Value GetDomRepresentation() const;
+        static AZ::Outcome<PatchOperation, AZStd::string> CreateFromDomRepresentation(Value domValue);
+
+        AZ::Outcome<PatchOperation, AZStd::string> GetInverse(Value stateBeforeApplication) const;
+
+    private:
+        struct PathContext
+        {
+            Value& m_value;
+            PathEntry m_key;
+        };
+
+        static constexpr AZ::u8 DefaultExistenceCheck = 0x0;
+        static constexpr AZ::u8 VerifyFullPath = 0x1;
+        static constexpr AZ::u8 AllowEndOfArray = 0x2;
```

Comment:
Why not use an `enum class` with the `AZ_DEFINE_ENUM_BITWISE_OPERATORS` macro to add in the bitwise comparison operators

Reply:
Good call, done.